### DISCUSSION TITLE
React Native & useInfiniteQuery

### DISCUSSION THREAD

**Discussion by egecavusoglu** [trpc/trpc#4236] [Q&A]
Is `useInfiniteQuery` unavailable in React Native? My procedure does have a cursor in the input object bu no luck.

**Comment by Shayan-retrocket** [comment]
Yes it was cut off in September 2021

However, there are alternative solutions you can explore to achieve infinite scrolling behavior in React Native. Here are a couple of options:

Pagination with FlatList: React Native provides the FlatList component, which supports pagination out of the box. You can implement infinite scrolling by fetching additional data when the user reaches the end of the list. You can manage the state of the data and implement pagination logic in your component.

Third-party libraries: There are several third-party libraries available that offer infinite scrolling functionality in React Native. One popular library is react-native-infinite-scrolling, which provides an easy way to implement infinite scrolling behavior in your app.

When using these solutions, you'll need to handle the pagination logic manually, including fetching the next set of data when the user reaches the end of the list, managing the cursor or page number, and updating the list accordingly.

Please note that React Native ecosystem evolves over time, and there might be new libraries or updates that provide direct support for infinite scrolling. It's always a good idea to check the latest documentation and community resources to find the most up-to-date solutions for your specific use case.

**Comment by egecavusoglu** [comment]
Thanks for the answer, we did implement with a regular query with pagination params, all is working fine. Cheers

**Reply by jeffngugi** [reply]
How did you implement such, any code snippet would be appreciated.
